pkg/util/clusterdata/worker_profile_test.go

```diff
@@ -21,260 +21,219 @@ import (
+
+	invalidProvSpec := machinev1beta1.ProviderSpec{Value: &kruntime.RawExtension{
+		Raw: []byte("invalid")}}
+
+	emptyProvSpec := machinev1beta1.ProviderSpec{}
+	noRawProvSpec := machinev1beta1.ProviderSpec{Value: &kruntime.RawExtension{}}
+
+	invalidWorkerProfile := []api.WorkerProfile{{Name: "fake-worker-profile-1", Count: 1}}
+	emptyWorkerProfile := []api.WorkerProfile{}
+
+	testCases := []struct {
+		name    string
+		client  machineclient.Interface
+		gotOc   *api.OpenShiftCluster
+		wantOc  *api.OpenShiftCluster
+		wantErr string
+	}{
+		{ //Test case: 1
+			name:   "machine set objects exists - valid provider spec JSON",
+			client: machinefake.NewSimpleClientset(createMachineSet("fake-worker-profile-1", validProvSpec()), createMachineSet("fake-worker-profile-2", validProvSpec())),
+			wantOc: getWantOc(validWorkerProfile()),
+			gotOc:  getGotOc(clusterID),
+		},
+		{ //Test case: 2
+			name:   "machine set objects exists - invalid provider spec JSON",
+			client: machinefake.NewSimpleClientset(createMachineSet("fake-worker-profile-1", invalidProvSpec)),
+			wantOc: getWantOc(invalidWorkerProfile),
+			gotOc:  getGotOc(clusterID),
+		},
+		{ //Test case: 3
+			name:   "machine set objects exists - provider spec is missing",
+			client: machinefake.NewSimpleClientset(createMachineSet("fake-worker-profile-1", emptyProvSpec)),
+			wantOc: getWantOc(invalidWorkerProfile),
+			gotOc:  getGotOc(clusterID),
+		},
+		{ //Test case: 4
+			name:   "machine set objects exists - provider spec is missing raw value",
+			client: machinefake.NewSimpleClientset(createMachineSet("fake-worker-profile-1", noRawProvSpec)),
+			wantOc: getWantOc(invalidWorkerProfile),
+			gotOc:  getGotOc(clusterID),
+		},
+		{ //Test case: 5
+			name:   "machine set objects do not exist",
+			client: machinefake.NewSimpleClientset(),
+			wantOc: getWantOc(emptyWorkerProfile),
+			gotOc:  getGotOc(clusterID),
+		},
+		{ //Test case: 6
+			name:   "machine set list request failed",
+			client: createFakeClientWithError(),
 			wantOc: &api.OpenShiftCluster{
 				ID: clusterID,
 			},
+			gotOc:   getGotOc(clusterID),
 			wantErr: "fake list error",
 		},
-		{
-			name: "invalid cluster object",
-			client: func() machineclient.Interface {
-				return machinefake.NewSimpleClientset()
-			},
-			modifyOc: func(oc *api.OpenShiftCluster) {
-				oc.ID = "invalid"
-			},
+		{ //Test case: 7
+			name:   "invalid cluster object",
+			client: machinefake.NewSimpleClientset(),
 			wantOc: &api.OpenShiftCluster{
 				ID: "invalid",
 			},
+			gotOc:   getGotOc("invalid"),
 			wantErr: "parsing failed for invalid. Invalid resource Id format",
 		},
-	} {
-		t.Run(tt.name, func(t *testing.T) {
-			oc := &api.OpenShiftCluster{
-				ID: clusterID,
-			}
-
-			if tt.modifyOc != nil {
-				tt.modifyOc(oc)
-			}
-
-			e := machineClientEnricher{}
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			//Given
+			// Create a clients object with the given client object
```

Comment:
super-nit: you'll see many differing opinions on this throughout your career: do you add comments to code, or not?

My impression is that this team tends to prefer the stance of "if the code is not clear and/or reading like english, and you need a comment to explain it, then your code is likely the problem, not the comment." We generally don't add comments, and instead let the code speak for itself. In rare cases where code is highly unclear where we needed to do something that is actually unusual and/or a head-scratcher, we might add comments referencing a known bug somewhere, or other additional information that explains _why_ something happened, rather than _what_ is happening.

I guess, to get to the point, you should be really proud of this test because these comments add nothing! You found a way to write really elegant and clear code that reads like english. I'd remove all the comments in this `t.Run` except the `Given/When/Then` comments that clearly define and add test structure. All those other comments are superfluous because your code speaks for itself, great work!

Reply:
made the changes accordingly